commerce/bookstore/python/load_data.py

```diff
@@ -0,0 +1,286 @@
+import csv, uuid, random
+from typedb.client import TypeDB, SessionType, TransactionType
+
+data_path = "data/"  # path to csv files to import/load data
+db = 'bookstore'  # Name of the DB to connect on the TypeDB
+
+
+def parse_data_to_dictionaries(input):
+    """
+      1. reads the file through a stream,
+      2. adds the dictionary to the list of items
+      :param input.file as string: the path to the data file, minus the format
+      :returns items as list of dictionaries: each item representing a data item from the file at input.file
+    """
+    #print('Parsing of ' + input["file"] + 'started.')
+    items = []
+
+    with open(input["file"] + ".csv", encoding='UTF-8') as data:  # 1
+        for row in csv.DictReader(data, delimiter=";", skipinitialspace=True):
+            item = {key: value for key, value in row.items()}  # fieldnames (keys) are taken from the first row
+            items.append(item)  # 2
+    print('Parsing of ' + input["file"] + ' successful.')
+    return items
+
+
+def load_data_into_typedb(input, session):
+    """
+      loads the csv data into our TypeDB phone_calls database:
+      1. gets the data items as a list of dictionaries
+      2. for each item dictionary
+        a. creates a TypeDB transaction
+        b. constructs the corresponding TypeQL insert query
+        c. runs the query
+        d. commits the transaction
+      :param input as dictionary: contains details required to parse the data
+      :param session: off of which a transaction will be created
+    """
+    items = parse_data_to_dictionaries(input)  # 1
+    for item in items:  # 2
+        with session.transaction(TransactionType.WRITE) as transaction:  # a
+            TypeQL_insert_query = input["template"](item)  # b # This calls one of the _template functions
+            #print("Executing TypeQL Query: " + TypeQL_insert_query)
+            transaction.query().insert(TypeQL_insert_query)  # c returns a list of answers
+            transaction.commit()  # d
+
+    print("Inserted " + str(len(items)) +
+          " items from [ " + input["file"] + ".csv] into TypeDB.\n")
+    return  # END of load_data_into_typedb()
+
+
+def books_template(book):  # building a TypeQL request to insert a book
```

Comment:
Right, so we've used a code comment here to clarify the meaning of the method, because it's not fully clear from the name alone. While that's better than not clarifying it at all, I'd say the ideal state is that the method is _already_ clear about its purpose just from its name.

As such, I would suggest renaming this (and the other related methods below) to something like `insert_book_query`. Then delete the comment.

Reply:
Fixed that in https://github.com/vaticle/typedb-examples/pull/143/commits/5f7a34bfa0797188c783c27fe3a9373671f1a23c